### PR TITLE
fix high performance hook self-exit container issue

### LINKED ISSUE #4095: high performance hook need to handle self-exit container
**Description**

if a container that uses the high performance hook to set up CPUSLoadBalancing exits by itself, rather than using kubectl delete command to terminate it, then the CPUSLoadBalancing doesn't get a chance to recover. For example, using the follow yaml file to create a cyclictest pod that lives for "5s", 
```
apiVersion: v1 
kind: Pod 
metadata:
  name: cyclictest 
  annotations:
    cpu-load-balancing.crio.io: "true"
spec:
  runtimeClassName: high-performance
  restartPolicy: Never 
  containers:
  - name: container-perf-tools 
    image: quay.io/jianzzha/perf-tools
    imagePullPolicy: IfNotPresent
    resources:
      limits:
        memory: "200Mi"
        cpu: "8"
    env:
    - name: tool
      value: "cyclictest"
    - name: DURATION
      value: "5s"
    - name: DISABLE_CPU_BALANCE
      value: "n"
    securityContext:
      privileged: true
    volumeMounts:
    - mountPath: /dev/cpu_dma_latency
      name: cstate
  volumes:
  - name: cstate
    hostPath:
      path: /dev/cpu_dma_latency
  nodeSelector:
    kubernetes.io/hostname: "worker0" 
```

After 5 second, the cyclictest completes and the pod state changes to "Completed". 

The cpu load balancing setting caused by the high-performance hooks is left behind and does not recover to its default value - it should revert back to 4655 after the pod is goes to "Completed" state. Deleting this "completed state" pod does not revert this setting.
```
/proc/sys/kernel/sched_domain/cpu0/domain0
4655
/proc/sys/kernel/sched_domain/cpu1/domain0
4655
/proc/sys/kernel/sched_domain/cpu10/domain0
4654
/proc/sys/kernel/sched_domain/cpu11/domain0
4654
/proc/sys/kernel/sched_domain/cpu12/domain0
4654
/proc/sys/kernel/sched_domain/cpu13/domain0
4655
/proc/sys/kernel/sched_domain/cpu14/domain0
4655
/proc/sys/kernel/sched_domain/cpu15/domain0
4655
/proc/sys/kernel/sched_domain/cpu16/domain0
4655
/proc/sys/kernel/sched_domain/cpu17/domain0
4655
/proc/sys/kernel/sched_domain/cpu18/domain0
4655
/proc/sys/kernel/sched_domain/cpu19/domain0
4655
/proc/sys/kernel/sched_domain/cpu2/domain0
4655
/proc/sys/kernel/sched_domain/cpu20/domain0
4655
/proc/sys/kernel/sched_domain/cpu21/domain0
4655
/proc/sys/kernel/sched_domain/cpu22/domain0
4655
/proc/sys/kernel/sched_domain/cpu23/domain0
4655
/proc/sys/kernel/sched_domain/cpu3/domain0
4655
/proc/sys/kernel/sched_domain/cpu4/domain0
4654
/proc/sys/kernel/sched_domain/cpu5/domain0
4654
/proc/sys/kernel/sched_domain/cpu6/domain0
4654
/proc/sys/kernel/sched_domain/cpu7/domain0
4654
/proc/sys/kernel/sched_domain/cpu8/domain0
4654
/proc/sys/kernel/sched_domain/cpu9/domain0
4654
```

**Steps to reproduce the issue:**
1. create a the cyclictest pog using above yaml or any yaml file that execution exits after certain time (for example centos with a sleep)
2. check the high performance hook set the cpu load balancing
3. after the pod is completed, check the cpu load balancing setting is same as in step2
4. kubectl delete the pod, check the cpu load balancing setting is same as in step2

**Describe the results you received:**
when the pod gets into "Completed" state, its cpu load balancing setting is not reverted

**Describe the results you expected:**
when the pod gets into "Completed" state, its cpu load balancing setting should be reverted

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by jianzzha**
Signed-off-by: Jianzhu Zhang<jianzzha@redhat.com>

#### What type of PR is this?

> /kind api-change
/kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
Bug fix
#### Which issue(s) this PR fixes:
Fixes #4095 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: server/sandbox_stop_linux.go [hunk 1]
```diff
@@ -9,9 +9,11 @@ import (
 	"github.com/cri-o/cri-o/internal/lib/sandbox"
 	"github.com/cri-o/cri-o/internal/log"
 	oci "github.com/cri-o/cri-o/internal/oci"
+	"github.com/cri-o/cri-o/internal/runtimehandlerhooks"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
 	"golang.org/x/sync/errgroup"
+
```
Comment (haircommander) [inline server/sandbox_stop_linux.go:16]: nit, can you drop this newline?

**Comment by haircommander** [comment]
/ok-to-test

one nit, otherwise LGTM, thanks @jianzzha

**Comment by TomSweeneyRedHat** [comment]
LGTM
but all kinds of test unhappiness

**Comment by haircommander** [comment]
hey @jianzzha you have a git validation problem:
https://app.circleci.com/pipelines/github/cri-o/cri-o/5512/workflows/f521b8b5-12e5-4880-ab8d-12ab9582716c/jobs/85710

**Comment by jianzzha** [comment]
@haircommander Thanks Peter! Do I need to worry about the test coverage?

**Comment by haircommander** [comment]
ideally, yes, but we don't block on it. We sure love unit tests though :smile:

**Comment by haircommander** [comment]
> @haircommander any suggestion on e2e_crun_cgroupv2 failure? The error doesn't seems to be relevant to this PR ...

it's non blocking, and sometimes flakes. don't worry about it :)

**Pull request merged**